File: AspNetMvcUnleashed/0672329980_source/Chapter10/VB/MvcApplication1/MvcApplication1/Controllers/DynamicController.vb
Public Class DynamicController
    Inherits Controller

    Public Function Index() As ActionResult
        ' Randomly select master page
        Dim rnd = New Random()
        Dim masterPage = "Dynamic" & rnd.Next(2)

        ' Return view with master page
        Return View("Index", masterPage)
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter10/CS/MvcApplication1/MvcApplication1/Controllers/DynamicController.cs
using System;
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class DynamicController : Controller
    {
        public ActionResult Index()
        {
            // Randomly select master page
            var rnd = new Random();
            var masterPage = "Dynamic" + rnd.Next(2);

            // Return view with master page
            return View("Index", masterPage);
        }

    }
}
